tests/e2e/utils/decorators.py

```diff
@@ -34,7 +34,7 @@ def wrapper(*args, **kwargs):
                     return test_function(*args, **kwargs)
 
                 except Exception as e:
-                    if type(e) != on_error:
+                    if not isinstance(e, on_error):
```

Comment:
aren't they working the same way?

Reply:
no, unlike a direct type comparison, `isinstance` will also check if an object
is an instance of a class or a subclass thereof. New Ruff is taking care of it (I have not found it personally ;)